internal/config/task_config.go

```diff
@@ -0,0 +1,88 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+const defaultConfigPath = "config/tasks.yaml"
+
+type TaskConfig struct {
+	Dependencies map[string]Dependency
+	Tasks        []Task `yaml:"tasks"`
+	ConfigPath   string
+}
+
+type Dependency struct {
+	Host string `yaml:"host"`
+	Port int    `yaml:"port"`
+	TLS  bool   `yaml:"tls"`
+}
+
+type Task struct {
+	Name    string `yaml:"name"`
+	Command string `yaml:"command"`
+}
+
+func (t *TaskConfig) MustLoad() {
+	if t.ConfigPath == "" {
+		t.ConfigPath = defaultConfigPath
+	}
+
+	yamlFile, err := ioutil.ReadFile(t.ConfigPath)
```

Comment:
Ah right - and yea os.Open would have been a better choice with the decoder, thanks!

Reply:
Also - a linter would have caught the ioutil depreciation ;) I know because I've seen it called out in the past on projects where we had linting configured.

Need to add that as a default for any project.